homeassistant/components/mqtt/models.py

```diff
@@ -91,17 +93,21 @@ def _convert_outgoing_payload(
 
             return payload
 
-        if self._attr_command_template is None:
+        if self._command_template is None:
             return value
 
-        values = {"value": value}
+        values: dict[str, Any] = {"value": value}
         if self._entity:
             values[ATTR_ENTITY_ID] = self._entity.entity_id
             values[ATTR_NAME] = self._entity.name
+            if self._command_template.hass:
```

Comment:
This must be an unnecessary check, is mypy complaining without it?

Reply:
it is, hass can be None

Reply:
Right, but it can't be none if `self._entity` is set which we already check?

Reply:
Removed it now since mypy is not complaining anymore